internal/blocktx/store/postgresql/postgres_test.go

```diff
@@ -611,3 +444,221 @@ func TestPostgresDB(t *testing.T) {
 		assert.Equal(t, expectedUnverifiedBlockHeights, res.UnverifiedBlockHeights)
 	})
 }
+
+func TestPostgresStore_UpsertBlockTransactions(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	tcs := []struct {
+		name        string
+		txs         []*blocktx_api.TransactionAndSource
+		merklePaths []string
+
+		expectedErr           error
+		expectedUpdatedResLen int
+	}{
+		{
+			name: "mismatched lengths of merkle paths and transactions",
+			txs: []*blocktx_api.TransactionAndSource{
+				{
+					Hash: revChainhash(t, "76732b80598326a18d3bf0a86518adbdf95d0ddc6ff6693004440f4776168c3b")[:],
+				},
+				{
+					Hash: revChainhash(t, "164e85a5d5bc2b2372e8feaa266e5e4b7d0808f8d2b784fb1f7349c4726392b0")[:],
+				},
+			},
+			merklePaths: []string{"test1"},
+			expectedErr: errors.New("transactions (len=2) and Merkle paths (len=1) have not the same lengths"),
+		},
+		{
+			name: "upsert all registered transactions (updates only)",
+			txs: []*blocktx_api.TransactionAndSource{
+				{
+					Hash: revChainhash(t, "76732b80598326a18d3bf0a86518adbdf95d0ddc6ff6693004440f4776168c3b")[:],
+				},
+				{
+					Hash: revChainhash(t, "164e85a5d5bc2b2372e8feaa266e5e4b7d0808f8d2b784fb1f7349c4726392b0")[:],
+				},
+			},
+			merklePaths:           []string{"test1", "test2"},
+			expectedUpdatedResLen: 2,
+		},
+		{
+			name: "upsert all non-registered transactions (inserts only)",
+			txs: []*blocktx_api.TransactionAndSource{
+				{
+					Hash: revChainhash(t, "edd33fdcdfa68444d227780e2b62a4437c00120c5320d2026aeb24a781f4c3f1")[:],
+				},
+			},
+			merklePaths:           []string{"test1"},
+			expectedUpdatedResLen: 0,
+		},
+		{
+			name: "update exceeds max batch size (more txs than 5)",
+			txs: []*blocktx_api.TransactionAndSource{
+				{
+					Hash: revChainhash(t, "b4201cc6fc5768abff14adf75042ace6061da9176ee5bb943291b9ba7d7f5743")[:],
+				},
+				{
+					Hash: revChainhash(t, "37bd6c87927e75faeb3b3c939f64721cda48e1bb98742676eebe83aceee1a669")[:],
+				},
+				{
+					Hash: revChainhash(t, "952f80e20a0330f3b9c2dfd1586960064e797218b5c5df665cada221452c17eb")[:],
+				},
+				{
+					Hash: revChainhash(t, "861a281b27de016e50887288de87eab5ca56a1bb172cdff6dba965474ce0f608")[:],
+				},
+				{
+					Hash: revChainhash(t, "9421cc760c5405af950a76dc3e4345eaefd4e7322f172a3aee5e0ddc7b4f8313")[:],
+				},
+				{
+					Hash: revChainhash(t, "8b7d038db4518ac4c665abfc5aeaacbd2124ad8ca70daa8465ed2c4427c41b9b")[:],
+				},
+			},
+			merklePaths:           []string{"test1", "test2", "test3", "test4", "test5", "test6"},
+			expectedUpdatedResLen: 6,
+		},
+	}
+
+	// common setup for test cases
+	ctx, _, sut := setupPostgresTest(t)
+	defer sut.Close()
+	sut.maxPostgresBulkInsertRows = 5
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			prepareDb(t, sut.db, "fixtures/update_block_transactions")
```

Comment:
I think we should rename the folder `fixtures/update_block_transactions` to `fixtures/upsert_block_transactions`

Reply:
Right